Fixed a bug where the formatResults function in the findHashed method was wiping out the resultset.

joshrhykerd opened the pull request:
I added a catchall condition that keeps `newResult` from becoming empty.

I ran into this issue on my own, but I believe it is a fix for https://github.com/dereuromark/cakephp-hashid/issues/11 and possibly also https://github.com/dereuromark/cakephp-hashid/issues/13

joshrhykerd committed:
Fixed a bug where the formatResults function in the findHashed method was wiping out the resultset.

I added a catchall condition that keeps newResult from becoming empty.

dereuromark commented:
Can we maybe have a Test  case  to make sure this doesnt happen again?
Thx for looking into this.

joshrhykerd committed:
Added testIsUniqueDomainRule

Added testIsUniqueDomainRule to guard against https://github.com/dereuromark/cakephp-hashid/issues/11 as requested.

dereuromark commented:
Thanks a lot!

joshrhykerd commented:
No problem!

dereuromark commented:
Do you happen to know  anything about the open PR for exists() and if it solves it?

joshrhykerd commented:
Based on this comment https://github.com/dereuromark/cakephp-hashid/issues/13#issuecomment-299849966 I would tend to think it would solve it as this was the issue with IsUniuque, but I can invenstigate it further if you'd like to be certain.

dereuromark commented:
That would be awesome! Then I can patch-release both fixes together.

joshrhykerd commented:
It seems that this fix does also resolve https://github.com/dereuromark/cakephp-hashid/issues/13 as we thougth it might. I provided a new test case to prove this.

dereuromark commented:
Yeah thats what I also thought, thanks for confirming it, I just released.
